diff --git a/server/Cargo.toml b/server/Cargo.toml
index cd36bd9c94..fa143b1f92 100644
--- a/server/Cargo.toml
+++ b/server/Cargo.toml
@@ -4,8 +4,8 @@ strip = true
 [workspace.dependencies]
 # dependencies used in graphql crates:
 # Warning don't add anything above this lines as auto patch is applied during docker build
-async-graphql = { version = "7.0.13", features = ["dataloader", "chrono"] }
-async-graphql-actix-web = "7.0.13"
+async-graphql = {git = "https://github.com/andreievg/async-graphql.git", rev = "6c476d50f0f2132860445b5ed5b3c6dd6e8cec92", version = "7.0.16", features = ["dataloader", "chrono"] }
+async-graphql-actix-web = {package = "async-graphql-actix-web", git = "https://github.com/andreievg/async-graphql.git",  version = "7.0.16", rev = "6c476d50f0f2132860445b5ed5b3c6dd6e8cec92" }
 actix-web = { version = "4.9.0", default-features = false, features = ["macros", "rustls"] }
 actix-multipart = "0.7.2"
 anymap = "0.12.1"
